### PR TITLE
Force reinstalling dependencies when install dir is empty

### PR DESCRIPTION
**Pull request by nmellado**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Update cmake scripts.

* **What is the current behavior?** (You can also link to an open issue here)
When Radium is already configured, cmake fails if the content of the install directory is empty, because dependencies cannot be found anymore.

* **What is the new behavior (if this is a feature change)?**
Detect when the content is removed by touching and checking the existence of dummy files in install dir. When these files cannot be found, the dependencies are re-installed.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: cmake/externalFunc.cmake [hunk 1]
```diff
@@ -89,4 +95,7 @@ macro(addExternalFolder NAME FOLDER )
     add_custom_target(External${NAME} ALL SOURCES ${external_sources})
     include("${CMAKE_CURRENT_BINARY_DIR}/external/cmake/package-${NAME}.cmake")
 
+    # Create touch file
+    file( TOUCH ${CMAKE_INSTALL_PREFIX}/share/${NAME}.touch )
```
Comment (MathiasPaulin) [inline cmake/externalFunc.cmake:99]:
```suggestion
    file( TOUCH "${CMAKE_INSTALL_PREFIX}/share/${NAME}.touch" )
```

File: cmake/externalFunc.cmake [hunk 2]
```diff
@@ -26,11 +26,17 @@ macro(addExternalFolder NAME FOLDER )
     # Check if install prefix has changed. If yes, force installing the externals again
     if( DEFINED CACHED_INSTALL_PREFIX )
         if( NOT "${CACHED_INSTALL_PREFIX}" STREQUAL "${CMAKE_INSTALL_PREFIX}" )
-            message(STATUS "[addExternalFolder] CMAKE_INSTALL_PREFIX has changed (from ${CACHED_INSTALL_PREFIX} to ${CMAKE_INSTALL_PREFIX}), reinstalling dependency")
+            message(STATUS "[addExternalFolder] CMAKE_INSTALL_PREFIX has changed (from ${CACHED_INSTALL_PREFIX} to ${CMAKE_INSTALL_PREFIX}), reinstalling dependencies")
             set( UPDATE_EXTERNAL ON)
         endif()
     endif()
 
+    # Check if install directory is empty. If yes, force installing the externals again
```
Comment (MathiasPaulin) [inline cmake/externalFunc.cmake:34]: This comment is not representative of what is done. Here, you search if one "external" (${NAME}) is installed.
```suggestion
    # Check if the external ${NAME} is installed. If not, force installing the externals again
```

**Comment by nmellado** [comment]
Todo: 
 

- [ ] Use `install` command instead of `file` to fix appveyor.

**Comment by nmellado** [comment]
Ready to review.

**Pull request merged**